src/abstractdataframe/subset.jl

```diff
@@ -107,6 +107,13 @@ described for [`select`](@ref) with the restriction that:
   `AbstractDataFrame`, `NamedTuple`, `DataFrameRow` or `AbstractMatrix`
   is not supported).
 
+The `subset` function is optimized for speed when low number of of conditions passed
+(up to several dozens). If you need to subset rows based on a larger number
+of columns prefer passing one condition using the syntax
+`AsTable(column_selector) => condition_function∘collect` when the condition function
+or for row-wise condition `AsTable(column_selector) => ByRow(condition_function∘collect)`,
+see [`DataFrames.table_transformation`](@ref) documentation for a detailed explanation.
```

Comment:
I'd put this in a `!!! note` and at the end of the docstring.

Also:
```suggestion
The `subset` function is optimized for speed when a low number of conditions are passed
(up to several dozens). If you need to subset rows based on a larger number
of columns prefer applying a single condition on all columns at once
using the syntax `AsTable(cols) => condition_function∘collect`
or for row-wise condition `AsTable(cols) => ByRow(condition_function∘collect)`,
see [`DataFrames.table_transformation`](@ref) documentation for a detailed explanation.
```

Reply:
I ended up adding a new code path.